### ISSUE TITLE
Warn about descriptors not in methods

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1033]
# Rule request

## Thesis

Descriptors do magic only as methods. Let's warn about using them for regular functions:

```python
@classmethod
def f():
    ...

@staticmethod
def f():
    ...

@property
def f():
    ...
```

## Reasoning

In most cases, it's a typo, and that function was intended to be a method.

**Comment by sobolevn** [comment]
Agreed!

**Comment by zteoh** [comment]
@sobolevn Hi, i'm trying to work on this and was wondering where would be the best location to have the test file and the visitor?

So far, I was able to:

1. Create a violation in best practices
```
@final
class MethodDecoratorUsedForFunction(ASTViolation):
    error_template = 'Method decorators (eg. classmethod, staticmethod and property) should not be used for functions'
    code = 450
```

2. Write some logic in the `ModuleMembersVisitor` class in `visitors/ast/complexity/counts.py`
```
def _check_method_decorators(self, node: ModuleMembers) -> None:
        parent_is_class = isinstance(get_parent(node), ast.ClassDef)

        # Find decorators that are of ast.Name and ignore ast.Call
        filtered_decorators = list(filter(lambda dec: isinstance(dec, ast.Name), node.decorator_list))
        # Check whether filtered decorators are method decorators
        is_method_decorator = list(map(lambda dec: dec.id in ['classmethod', 'staticmethod', 'property'], filtered_decorators))
        # Check for presence of method deecorators
        has_method_decorators = reduce(lambda x,y:x or y, is_method_decorator, False)

        if has_method_decorators and not parent_is_class:
            self.add_violation(
                MethodDecoratorUsedForFunction(node),
            )
```
Thank you so much!
